### ISSUE TITLE
[a11y | SelectDialog]: headerText is not read by VoiceOver

### ISSUE THREAD

**Issue by volha-arkhipenka-sap** [SAP/ui5-webcomponents-react#7319]
### Bug Description

With Voiceover enabled, on opening `SelectDialog` the title provided in `headerText` is not read by VoiceOver. 

### Affected Component

`SelectDialog` https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/modals-popovers-selectdialog--docs

### Expected Behaviour

All `SelectDialog` content and information should be conveyed to a user.

### Isolated Example

Reproducible with VoiceOver enabled on https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/modals-popovers-selectdialog--docs

### Steps to Reproduce

1. Provide `headerText` prop to `SelectDialog` component
2. Enable VoiceOver
3. Open `SelectDialog` component
4. `headerText` is not announced

### Log Output, Stack Trace or Screenshots

<img width="1724" alt="Image" src="https://github.com/user-attachments/assets/11371875-5d50-47a1-8f30-f8c000b1eb20" />

### Priority

None

### UI5 Web Components Version

2.9.0

### Browser

Chrome

### Operating System

macOS Sequoia 15.4.1

### Declaration

- [x] I’m not disclosing any internal or sensitive information.
